src/catalog/src/system_schema/information_schema/procedure_info.rs

```diff
@@ -177,8 +177,8 @@ impl InformationSchemaProcedureInfoBuilder {
                         .map_err(BoxedError::new)
                         .context(ListProceduresSnafu)?;
                     for procedure in procedures {
-                        let status = procedure.state.as_str_name();
-                        self.add_procedure(&predicates, procedure, status);
+                        let status = procedure.state.as_str_name().to_string();
```

Comment:
Em... Well, the `status` will be cloned twice...

Reply:
Ok,  how about change `add_procedure`  from `fn add_procedure(&mut self, predicates: &Predicates, procedure_info: ProcedureInfo, status: &str)` to `fn add_procedure(&mut self, predicates: &Predicates, status: String, procedure_info: ProcedureInfo)`

Reply:
Yes, that is what I mean.